Documentation/Guides/SSL Pinning.md

```diff
@@ -0,0 +1,36 @@
+
+For example, let's assume we want to pin all our connections to https://jsonplaceholder.typicode.com. At the time the app connects to that TLS network, the server will provide it's SSL certificate for validation. When pinning is enabled, the app will inspect the contents of the certificate, and if that certificate does not match one that it is looking for (even if the certificate is valid and trusted by iOS) the request will be rejected.
+
+The first thing we need to do then, is create a certificate to pin to. Assuming you have `certificate.cer` for this domain, you can pass this certificate directly to the `TrustConfiguration` using the `DomainConfiguration(domain:certificates:)`. This initializer will automatically create a formatted hash for this certificate and use that as the basis for pinning.
+
+If you would like, instead of embedding certificates directly into the bundle, you can manually create this hash. A SHA-256 hash of the public key of the certificate is used by the `TrustValidator` under the hood. Again, using the `jsonplaceholder.typicode.com` example - you can use a variety of tools to view the details of the certificate itself (include Safari - click the padlock in the address bar). This will allow you to see the hex bytes of the public key of the certificate. At the time of writing, these are:
+
+`04 B9 ED ED FC C4 3D 5D 2B C6 64 C1 53 A3 B6 52 56 7B BF E2 86 AB FF 94 97 FD 9C CA 11 3B 23 2D 8B 74 7B 94 7F 99 00 13 FA D1 01 73 00 B3 90 3B 74 F2 20 D7 47 53 D4 F8 33 58 A5 07 F5 12 39 8A 7E`
+
+These bytes can then be run through a SHA-256 hash (an easy way to do this is using the `CertificateHasher` in Hyperspace) to produce your pinning hash: `XlCPXC6IrttTF9Y1887kS+efCCf3uFjHW6D1TUI9f+Q=`. This hash can then be passed into a `DomainConfiguration` using `.init(domain:pinningHashes:)` or `.init(domain:encodedPinningHashes:)` (the latter of which is Base64 encoded). Please note that the hash required for the pinning implementation in Hyperspace is not identical to the one that the `NSPinnedDomains` key uses in iOS 14+.
+
+Once the `TrustConfiguration` has been created with an `[DomainConfiguration]`, the final step is to tell Hyperspace how to validate this configuration at request-time. This can be done by manually utilizing a `TrustValidator` at the appropriate time, and then by calling `validator.handle(challenge:,handler:)`. In the vast majority of cases though, it is simpler to use the provided `TrustValidatingTransportService` which builds upon `TransportService` to automatically provide this functionality when the app receives an authentication challenge through the `URLSessionDelegate` for an appropriate domain.
```

Comment:
Should be "created with a `[DomainConfiguration]`," instead of "created with an `[DomainConfiguration]`"

Reply:
I was going back on forth on that, between 'an array of `DomainConfiguration`' and 'a `DomainConfiguration` array'.  Will update.